package/network/utils/ltq-dsl-base/files/lib/functions/lantiq_dsl.sh

```diff
@@ -728,11 +742,52 @@ profile() {
 	fi
 }
 
+line_features() {
+	local trellisd
+	local bitswapd
+	local ginpd
+	local vnoised
+	
+	local trellisu
+	local bitswapu
+	local ginpu
+	local vnoiseu
+	
+	lfcgd=$(dsl_cmd lfcg 1 0)
```

Comment:
The first arg corresponds to
```
   DSL_MODE_ADSL = 0,
   /**
   VDSL mode selected.
   \note This option is only supported for VDSL capable platforms. */
   DSL_MODE_VDSL = 1,
```
Which raises the question if this works on an adsl line?

Reply:
Well spotted. I haven't tested this as I don't have adsl. Looking at other commands here they use a 0 as first argument most of the time (which is then adsl mode, but this certainly works on vdsl)?

Reply:
The first 0 at other commands is often the `nChannel` parameter.

I think you are generally confusing things here.
With `lfcg` you read the configured values (depending on the DSL mode). What you probably want to have here is the current status. And for this you can use the command `lfsg`. It has only one parameter for the direction (upstream/downstream) like most others.

Reply:
change done.